pages/en/developing/creating-a-subgraph.mdx

```diff
@@ -25,12 +25,36 @@ The `<SUBGRAPH_SLUG>` is the ID of your subgraph in Subgraph Studio, it can be f
 
 The second mode `graph init` supports is creating a new project from an example subgraph. The following command does this:
 
-```
+```sh
 graph init --studio <SUBGRAPH_SLUG>
 ```
 
 The example subgraph is based on the Gravity contract by Dani Grant that manages user avatars and emits `NewGravatar` or `UpdateGravatar` events whenever avatars are created or updated. The subgraph handles these events by writing `Gravatar` entities to the Graph Node store and ensuring these are updated according to the events. The following sections will go over the files that make up the subgraph manifest for this example.
 
+## Add New dataSources To An Existing Subgraph
+
+Since `v0.31.0` the `graph-cli` supports adding new dataSources to an existing subgraph through the `graph add` command.
+
+```sh
+graph add <address> [<subgraph-manifest default: "./subgraph.yaml">]
+
+Options:
+
+      --abi <path>              Path to the contract ABI (default: download from Etherscan)
+      --contract-name           Name of the contract (default: Contract)
+      --merge-entities          Whether to merge entities with the same name (default: false)
+      --network-file <path>     Networks config file path (default: "./networks.json")
+```
+The `add` command will fetch the ABI from Etherscan (unless an ABI path is specified with the `--abi` option), and will create a new  `dataSource` in the same way that `graph init` command creates a `dataSource` `--from-contract`, updating the schema and mappings accordingly.
```

Comment:
```suggestion
```

The `add` command will fetch the ABI from Etherscan (unless an ABI path is specified with the `--abi` option), and will create a new  `dataSource` in the same way that `graph init` command creates a `dataSource` `--from-contract`, updating the schema and mappings accordingly.
```